Objective-C/CBLMutableArray.mm

```diff
@@ -228,14 +241,47 @@ - (void) insertDictionary: (nullable CBLDictionary*)value atIndex: (NSUInteger)i
 
 #pragma mark - Set Content with an Array
 
-- (void) setData: (nullable NSArray*)data {
+- (void) setData: (NSArray*)data {
     CBL_LOCK(self.sharedLock) {
         _array.clear();
         for (id obj in data)
             _array.append([obj cbl_toCBLObject]);
     }
 }
 
+#pragma mark - SetJSON
+
+- (BOOL) setJSON: (NSString*)json error: (NSError**)error {
+    CBLStringBytes jsonSlice(json);
+    FLError flEerror = {};
+    Encoder enc;
+    if (!FLEncoder_ConvertJSON(enc, jsonSlice)) {
+        flEerror = enc.error();
+        CBLWarnError(Database, @"%@: Error converting JSON %d", self, flEerror);
+        convertError(flEerror, error);
+        return NO;
+    }
+    
+    flEerror = {};
+    FLSliceResult result = FLEncoder_Finish(enc, &flEerror);
+    if (!result.buf) {
+        CBLWarnError(Database, @"%@: Error decoding JSON: %d", self, flEerror);
+        convertError(flEerror, error);
+        return NO;
+    }
+    
+    FLArray array = FLValue_AsArray(FLValue_FromData(C4Slice(result), kFLTrusted));
+    CBL_LOCK(self.sharedLock) {
+        _array.clear();
```

Comment:
Can we just replace _array with a new MArray? There is `MArray(MValue *mv, MCollection *parent)` that can be used to create a new Array with `array` and the parent from `_array.parent`.

Reply:
https://issues.couchbase.com/browse/CBL-1760